pkg/handlers/host.go

```diff
@@ -107,9 +108,19 @@ func (h *HostHandlers) PatchHostByID(w http.ResponseWriter, req *http.Request) e
 	}
 	hostToDB, err := h.constructHostForDB(createHostRequest, req)
 	if err != nil {
-		return err
+		return api.WriteJSON(w, http.StatusInternalServerError, err.Error())
```

Comment:
To keep things standard for @diegocarhuavilca , we may want to use our error struct here:

```suggestion
		return api.WriteJSON(w, http.StatusInternalServerError, api.APIError{Error: err.Error()}))
```

Also, it is a good practice to log the error with `slog.Error()` before sending it back to the client.

PD:
As a side note, perhaps its time we write a constructor for this kind of error in the `api` package.. maybe a `api.WriteError(w http.Response, status int, message string)` that writes a JSON version of the error ? This could also allow us to write less _back-end_ specific errors to this struct (e.g: "unkown column 'ip' in 'host', and instead more user-friendly errors (e.g: 'An internal server error occurred').

Reply:
This constructor will be in the pr of mistakes 🥺